src/operation/decision_model.jl

```diff
@@ -351,14 +351,49 @@ results = solve!(OpModel, output_dir = "output")
 function solve!(
     model::DecisionModel{<:DecisionProblem};
     export_problem_results = false,
+    console_level = Logging.Error,
+    file_level = Logging.Info,
+    disable_timer_outputs = false,
+    serialize = true,
     kwargs...,
 )
-    status = solve_impl(model; kwargs...)
-    set_run_status!(model, status)
-    if status == RunStatus.SUCCESSFUL
-        results = ProblemResults(model)
-        serialize_results(results, get_output_dir(model))
-        export_problem_results && export_results(results)
+    build_if_not_already_built!(
+        model;
+        console_level = console_level,
+        file_level = file_level,
+        disable_timer_outputs = disable_timer_outputs,
+        kwargs...,
+    )
+    set_console_level!(model, console_level)
+    set_file_level!(model, file_level)
+    TimerOutputs.reset_timer!(RUN_OPERATION_MODEL_TIMER)
+    disable_timer_outputs && TimerOutputs.disable_timer!(RUN_OPERATION_MODEL_TIMER)
+    logger = configure_logging(model.internal, "a")
+    status = RunStatus.FAILED
+    try
+        Logging.with_logger(logger) do
+            TimerOutputs.@timeit RUN_OPERATION_MODEL_TIMER "Solve" begin
+                status = solve_impl(model; kwargs...)
+                set_run_status!(model, status)
+            end
+            if status == RunStatus.SUCCESSFUL
+                if serialize
+                    TimerOutputs.@timeit RUN_OPERATION_MODEL_TIMER "Serialize" begin
+                        optimizer = get(kwargs, :optimizer, nothing)
+                        serialize_problem(model, optimizer = optimizer)
+                        serialize_optimization_model(model)
+                    end
```

Comment:
@jd-lara @claytonpbarrows Note this change. This is the best way to ensure that we serialize the correct optimizer attributes.

And it only runs when the solve is successful. Is that correct? (Same applies for EmulationModel)